dvc/project.py

```diff
@@ -0,0 +1,203 @@
+import os
+import itertools
+import networkx as nx
+
+from dvc.logger import Logger
+from dvc.exceptions import DvcException
+from dvc.stage import Stage, Output, Dependency
+from dvc.config import Config
+from dvc.state import State
+from dvc.lock import Lock
+from dvc.scm import SCM
+from dvc.cache import Cache
+from dvc.data_cloud import DataCloud
+
+
+class PipelineError(DvcException):
+    pass
+
+
+class StageNotInPipelineError(PipelineError):
+    pass
+
+
+class StageNotFoundError(DvcException):
+    pass
+
+
+class Pipeline(object):
```

Comment:
Please consider extraction of this class into a separate file?

Reply:
Will do.